File: Attendance/Models/Request.cs
using Attendance.Models.SharedProp;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace Attendance.Models
{
    public class Request : Audit
    {
        public int Id { get; set; }
        public string TypeOfAbsence { get; set; }
        public DateTime From { get; set; }
        public DateTime To { get; set; }
        public string ReasonOfAbsence { get; set; }
        public long ManagerStatus { get; set; } = 3;
        public long HRStatus { get; set; } = 3;
        public int UserId { get; set; }
        public User Users { get; set; }
    }
}

File: Attendance/Angular/src/app/shared/models/request.ts
export class Request {
    Id: number;
    TypeOfAbsence?: number;
    From?: Date | string;
    To?: Date | string;
    ReasonOfAbsence?: string;
    ManagerStatus?: string;
    HRStatus?: string;
    UserId?: number;
    FilePath?: string;
    Name? : string;
  
    
    constructor(
      Id: number,
      TypeOfAbsence?: number,
      From?: Date,
      To?: Date,
      ReasonOfAbsence?: string,
      ManagerStatus?: string,
      HRStatus?: string,
      UserId?: number,
      FilePath?: string,
      Name? : string
    ) {
      this.Id = Id;
      this.TypeOfAbsence = TypeOfAbsence;
      this.From = From;
      this.To = To;
      this.ReasonOfAbsence = ReasonOfAbsence;
      this.ManagerStatus = ManagerStatus;
      this.HRStatus = HRStatus;
      this.UserId = UserId;
      this.FilePath = FilePath;
      this.Name = Name; 
    }
  }
  